src/examples/mfem_mortar_lm_patch.cpp

```diff
@@ -0,0 +1,293 @@
+  vi_dc.RegisterField("pos", &coords);
+
+  // grid function for displacement
+  mfem::ParGridFunction u { &par_fe_space };
+  pv_dc.RegisterField("disp", &u);
+  vi_dc.RegisterField("disp", &u);
+  u = 0.0;
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to create grid functions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // save initial configuration
+  pv_dc.Save();
+  vi_dc.Save();
+
+  // recover dirichlet bc tdof list
+  timer.start();
+  mfem::Array<int> ess_tdof_list;
+  {
+    mfem::Array<int> ess_vdof_marker;
+    mfem::Array<int> ess_bdr(pmesh->bdr_attributes.Max());
+    ess_bdr = 0;
+    for (auto xfix_attrib : xfix_attribs)
+    {
+      ess_bdr[xfix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, ess_vdof_marker, 0);
+    mfem::Array<int> new_ess_vdof_marker;
+    ess_bdr = 0;
+    for (auto yfix_attrib : yfix_attribs)
+    {
+      ess_bdr[yfix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, new_ess_vdof_marker, 1);
+    for (int i{0}; i < ess_vdof_marker.Size(); ++i)
+    {
+      ess_vdof_marker[i] = ess_vdof_marker[i] || new_ess_vdof_marker[i];
+    }
+    ess_bdr = 0;
+    for (auto zfix_attrib : zfix_attribs)
+    {
+      ess_bdr[zfix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, new_ess_vdof_marker, 2);
+    for (int i{0}; i < ess_vdof_marker.Size(); ++i)
+    {
+      ess_vdof_marker[i] = ess_vdof_marker[i] || new_ess_vdof_marker[i];
+    }
+    mfem::Array<int> ess_tdof_marker;
+    par_fe_space.GetRestrictionMatrix()->BooleanMult(ess_vdof_marker, ess_tdof_marker);
+    mfem::FiniteElementSpace::MarkerToList(ess_tdof_marker, ess_tdof_list);
+  }
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to set up boundary conditions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set up mfem elasticity bilinear form
+  timer.start();
+  mfem::ParBilinearForm a(&par_fe_space);
+  mfem::ConstantCoefficient lambda(50.0);
+  mfem::ConstantCoefficient mu(50.0);
+  a.AddDomainIntegrator(new mfem::ElasticityIntegrator(lambda, mu));
+  a.Assemble();
+
+  // compute elasticity contribution to stiffness
+  auto A = std::make_unique<mfem::HypreParMatrix>();
+  a.FormSystemMatrix(ess_tdof_list, *A);
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to create and assemble internal stiffness: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set up tribol
+  timer.start();
+  tribol::initialize(pmesh->SpaceDimension(), MPI_COMM_WORLD);
+  tribol::registerMfemMesh(
+    0, 0, 1, *pmesh, coords, mortar_attribs, nonmortar_attribs,
+    tribol::SURFACE_TO_SURFACE, 
+    tribol::NO_SLIDING, 
+    tribol::SINGLE_MORTAR, 
+    tribol::FRICTIONLESS,
+    tribol::LAGRANGE_MULTIPLIER,
+    tribol::BINNING_GRID
+  );
+  tribol::setLagrangeMultiplierOptions(
+    0,
+    tribol::ImplicitEvalMode::MORTAR_RESIDUAL_JACOBIAN
+  );
+
+  // update tribol (compute contact contribution to force and stiffness)
+  double dt {1.0};
```

Comment:
Arbitrary dt because this is simply a single solve example. Note for clarity?

Reply:
Done